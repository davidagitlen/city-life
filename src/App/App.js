import React from 'react';
import './App.scss';
import citySkyline from '../images/city-skyline.svg';
import CityForm from '../CityForm/CityForm';
import City from '../City/City';
import Comparison from '../Comparison/Comparison';
import Details from '../Details/Details';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import { NavLink, Route } from 'react-router-dom';


export const App = (props) => {
    console.log('in app', props)
    const { cityInfo } = props;
    const cityOneReady = cityInfo.one.scores.length && cityInfo.one.details && cityInfo.one.images.images;
    const cityTwoReady = cityInfo.two.scores.length && cityInfo.two.details && cityInfo.two.images.images;
    return (
      <div className='App'>
        <header className='App-header'>
          <div className='site-name'>
            <img src={citySkyline} alt='' />
            <h1>City<span>Life</span></h1>
          </div>
          <div className='nav-links'>
            <NavLink to='/' className='nav'>Compare Cities</NavLink>
            <NavLink to='/past-comparisons'>History</NavLink>
            <NavLink to='/world-map'>Map</NavLink>
          </div>
        </header>
        <main>
        <Route exact path='/' render={() => {
          return(
            <>
              <div className='forms'>
                <div className='city-one'>
                  <CityForm ordinal='one' />
                  {(cityOneReady || null) && <City ordinal='one' />}
                </div>
                <div className='city-two'>
                  <CityForm ordinal='two' />
                  {(cityTwoReady || null) && <City ordinal='two' />}
                </div>
              </div>
              <Comparison />
            </>
          )
        }} />
        <Route path='/details/:name' render={({ match }) => {
          const { name } =match.params;
          const awaitData = cityOneReady || cityTwoReady; 
          const cityCheck = awaitData ?cityInfo.one.details.fullName.includes(name) : null;
          const cityProps = cityCheck ? cityInfo.one : cityInfo.two;
          return <Details props={cityProps}/>
        }}
        />
        </main>
        {/* <Route exact path='/past-comparisons' render={() => {
          return(
            <main>
              <div>
                <Details />
              </div>
            </main>
          )
        }} /> */}
      </div>
    );
}

export const mapStateToProps = state => ({
  cityInfo: state.cityInfo
})

export default connect(mapStateToProps)(App);
