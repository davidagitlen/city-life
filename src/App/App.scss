.App {
  text-align: center;
}

.App-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 100px;
  width: 100%;
  background-color: ghostwhite;
  border: 1px solid blue;

  & div {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    height: 100%;
    width: 50%;
  }

  & h1 {
    font-size: 34px;
    font-weight: bold;
    position: relative;

    & span {
      color: blue;
      z-index: 1;
    }
  }

  & img {
    position: absolute;
    height: 260px;
    width: 400px;
    top: -70px;
  }
}

main {
  display: flex;
  flex-direction: row;
  background-color: blue;
  height: 100vh;
  width: 100%;
}

.forms {
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 33%;
}


// .App-link {
//   color: #61dafb;
// }

// @keyframes App-logo-spin {
//   from {
//     transform: rotate(0deg);
//   }
//   to {
//     transform: rotate(360deg);
//   }
// }
