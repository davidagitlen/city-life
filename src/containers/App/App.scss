.App {
  text-align: center;
}

.App-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 100px;
  width: 100%;
  background-color: #2E2EC9;

  & div {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    height: 100%;
    width: 50%;
  }

  & h1 {
    color: snow;
    font-size: 46px;
    position: relative;

    & span {
      font-weight: bold;
      color: snow;
      z-index: 1;
    }
  }

  & img {
    position: absolute;
    height: 260px;
    width: 400px;
    top: -70px;
  }
}

main {
  display: flex;
  flex-direction: row;
  height: 100vh;
  width: 100%;
}

.forms {
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 33%;
}

.placeholder {
  position: relative;
  height: 280px;
  width: 90%;
  background-color: ghostwhite;

   & img {
    height: 225px;
    margin: 15px 0px 15px 0px;
    position: relative;
    max-width: 100%;

    &#circle {
      -webkit-clip-path: circle(75px at 50% 50%);
      clip-path: circle(80px at 50% 50%);
      background-color: white;
    }
  }
}

.nav {
  color: snow;
  text-decoration: none;
  font-size: 1em;
  transition: font-size .3s ease-in-out;

  &:hover {
    color: #6DECAF;
    font-size: 1.2em;
    transition: font-size .3s ease-in-out;
  }

  &:focus {
    color: #6DECAF;
    outline: none;

  }
}

