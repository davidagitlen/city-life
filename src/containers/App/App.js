import React from 'react';
import './App.scss';
import citySkyline from '../../images/city-skyline.svg';
import CityForm from '../CityForm/CityForm';
import City from '../City/City';
import Comparison from '../Comparison/Comparison';
import Details from '../../components/Details/Details';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { NavLink, Route } from 'react-router-dom';

export const App = (props) => {
  const { cityInfoReducer: cityData } = props;
  const {
    inFlightScores, inFlightDetails, inFlightImages,
    details, scores,
  } = cityData;
  const isReady = !(inFlightScores.includes(true) || inFlightDetails.includes(true) || inFlightImages.includes(true));
  const findSelectedIndex = (dataSet, matchString) => (
    dataSet.details.findIndex(thing => thing.fullName.split(',').includes(matchString)));
  const formSet = [
    { ordinal: 0, selectedClass: 'city-one' },
    { ordinal: 1, selectedClass: 'city-two' }
  ];
  
    // const { scores: scoresOne = [], images: imagesOne = {}, } = one;
    // const { scores: scoresTwo = [] } = two;
    // const { images: imagesTwo = [] } = two;
    // const cityOneReady = (scoresOne.length > 0) && one.details && imagesOne.images;
    // const cityTwoReady = (scoresTwo.length > 0) && two.details && imagesTwo.images;

    return (
      <div className='App'>
        <header className='App-header'>
          <div className='site-name'>
            <h1>City<span>Life</span></h1>
          </div>
          <div className='nav-links'>
            <NavLink to='/' className='nav'>Compare Cities</NavLink>
            <NavLink to='/past-comparisons' className='nav'>History</NavLink>
            <NavLink to='/world-map' className='nav'>Map</NavLink>
          </div>
        </header>
        <main>
        <Route exact path='/' render={() => {
          return(
            <>
              <div className='forms'>
                {
                  formSet.map(({ ordinal, selectedClass }) => (
                  <div 
                    className={selectedClass}
                    key={selectedClass}
                    >
                  <CityForm ordinal={ordinal}/>
                  {
                    inFlightImages[ordinal] ?
                    <div className='placeholder'>
                      <img
                        id='circle'
                        src={citySkyline}
                        alt=''
                      />
                    </div> :
                    <City ordinal={ordinal} />
                  }
                </div>
                ))
                }
              </div>
              <Comparison />
            </>
          )
        }} />
        <Route
          path='/details/:name'
          render={({ match }) => {
              const { name } = match.params;
              
              if (isReady) {
                const selectedIndex = findSelectedIndex(cityData, name);

                return <Details cityData={cityData} index={selectedIndex}/>
              }
              return null;
            }}
        />
        </main>
      </div>
    );
}

export const mapStateToProps = state => ({
  ...state
})

export default connect(mapStateToProps)(App);

App.propTypes = {
  cityInfo: PropTypes.object,
  dispatch: PropTypes.func
}