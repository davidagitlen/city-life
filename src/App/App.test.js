import React from 'react';
import { mapStateToProps, App } from './App';
import { shallow } from 'enzyme';


describe('App', () => {

  let wrapper;
  let mockCityInfo;

  beforeEach(() => {
    mockCityInfo = {
      one: {
        details : {
          population: 1,
          latitude: 10,
          longitude: 10,
          fullName: 'Denver, Colorado!',
          timeZone: 'Denver!'
        },
        images: {
          attribution: {},
          images: {
            mobile: 'http://blurg.com',
            web: 'http://blurg.com/web'
          }
        },
        scores : [{color: 'blue', name: 'Score', score_out_of_10: 10}]
      },
      two: {
        details: {
          population: 2,
          latitude: 20,
          longitude: 20,
          fullName: 'Boulder, Colorado!',
          timeZone: 'Denver!'
        },
        images: {
          attribution: {},
          images: {
            mobile: 'http://blurg.com',
            web: 'http://blurg.com/web'
          }
        },
        scores: [{ color: 'red', name: 'Score', score_out_of_10: 10 }]
      }
    };

    wrapper = shallow(
      <App 
        cityInfo={mockCityInfo}
      />)
  });

  it('should match the snapshot with data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {

    it('should return an object with appropriate data', () => {
      
      const mockState = {
        cityInfo: mockCityInfo,
        unrealData: 'boo!'
      };

      const expected = {
        cityInfo: mockCityInfo,
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);

    })


  });
});

