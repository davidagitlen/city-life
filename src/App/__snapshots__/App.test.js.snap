// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with data passed in correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="site-name"
    >
      <img
        alt=""
        src="city-skyline.svg"
      />
      <h1>
        City
        <span>
          Life
        </span>
      </h1>
    </div>
    <div
      className="nav-links"
    >
      <NavLink
        className="nav"
        to="/"
      >
        Compare Cities
      </NavLink>
      <NavLink
        to="/past-comparisons"
      >
        History
      </NavLink>
      <NavLink
        to="/world-map"
      >
        Map
      </NavLink>
    </div>
  </header>
  <main>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
  </main>
</div>
`;
